### PR TITLE
Fix tests and prepare modern Ember compatibility

### PR DESCRIPTION
**Pull request by halfbyte**
Okay, here we go. This should result in green checks on gh actions, at least it did on a PR I opened against my own main branch.

This PR does:
- Minimal updates/changes to the addon dependencies, mainly for testing, so that we can get modern tests to run
- Convert all tests to modern module syntax
- Update ember-try config to
  - Get rid of the ember-cli-uglify scenarios
  - Test against all LTS versions down to 3.16
  - Prepare testing against 4.x releases
- Update the dummy test app to modern syntax as minimally as possible

In next steps here are some things I would like to deliver:
- update eslint config to modern ember standards including an up to date prettier config
- Convert the template compiler to the new functional syntax so that it will be 4.x compatible
- If at all possible, apply the other changes we did on our Focusrite-Novation fork as laid out in #102

Thanks @calvinlough for looking into this, I hope this PR has a size that is reviewable for you.

### CODE REVIEW

File: ember-cli-build.js [hunk 1]
```diff
@@ -1,8 +1,12 @@
 /* global require, module */
-var EmberApp = require('ember-cli/lib/broccoli/ember-addon');
+const EmberApp = require('ember-cli/lib/broccoli/ember-addon');
 
 module.exports = function(defaults) {
-  var app = new EmberApp(defaults, {});
+  const app = new EmberApp(defaults, {
+    'ember-cli-babel': {
+      includePolyfill: true
```
Comment (calvinlough) [inline ember-cli-build.js:7]: Is the polyfill actually required?

Comment (halfbyte) [reply ember-cli-build.js:7]: I'm pretty sure it was needed somewhere down the line - I'll try to get rid of it for this PR, it was a copy and paste thing.

Comment (halfbyte) [reply ember-cli-build.js:7]: @calvinlough Ah, I remember now, if I remove it, the async await in the tests and in ember-try won't work. I looked at the ember-addon blueprints and nothing seems to suggest that this needed though - Are you aware of a better way of fixing this?

Comment (calvinlough) [reply ember-cli-build.js:7]: That's unusual, I'm not sure why that's happening. It seems like this change would force the babel polyfill on anyone using the addon, which is a no-go. Do you see that error if you just run `yarn test` locally or is it a GH actions issue?

Comment (halfbyte) [reply ember-cli-build.js:7]: This might be an alternative that's better suited for an addon? 

https://github.com/machty/ember-maybe-import-regenerator

Comment (calvinlough) [reply ember-cli-build.js:7]: Sounds good

**Review by calvinlough — APPROVED** [review]

**Pull request merged**